edb/lang/edgeql/compiler/expr.py

```diff
@@ -82,6 +80,21 @@ def compile_BinOp(
     return setgen.ensure_set(op_node, ctx=ctx)
 
 
+@dispatch.compile.register(qlast.IsOp)
+def compile_IsOp(
+        expr: qlast.Base, *, ctx: context.ContextLevel) -> irast.Set:
+    try_folding = True
+
+    op_node = compile_type_check_op(expr, ctx=ctx)
+
+    if try_folding:
```

Comment:
This check is always `True`, remove the `if`.

Reply:
OK.